Lib/contextlib.py

```diff
@@ -388,49 +392,27 @@ def pop_all(self):
         self._exit_callbacks = deque()
         return new_stack
 
-    def _push_cm_exit(self, cm, cm_exit):
-        """Helper to correctly register callbacks to __exit__ methods"""
-        def _exit_wrapper(*exc_details):
-            return cm_exit(cm, *exc_details)
-        _exit_wrapper.__self__ = cm
-        self.push(_exit_wrapper)
-
     def push(self, exit):
         """Registers a callback with the standard __exit__ method signature
-
-        Can suppress exceptions the same way __exit__ methods can.
-
+        Can suppress exceptions the same way __exit__ method can.
```

Comment:
Please keep and empty line after the first line of the docstring.  I also recommend you to read https://www.python.org/dev/peps/pep-0257/.

Reply:
@1st1 In that PEP there is no empty line after a doctstring inside `def`. Should I add this extra empty line for all functions in my PR?

Reply:
Ilya, I mean this:

```python
def foo():
    """First line
    A paragraph
    """
    code
```

should be formatted as this

```python
def foo():
    """First line.

    A paragraph.
    """
    code
```

Reply:
The first line should contain the complete sentences.

I'm not making these rules up -- there're tons of tools out there that expect this specific code style to extract/parse docstrings.

Reply:
Ah I see.